app/src/main/java/com/github/geohunt/app/model/database/firebase/FirebaseDatabase.kt

```diff
@@ -29,18 +28,18 @@ class FirebaseDatabase(activity: Activity) : Database {
 
     // Database references
     internal val dbChallengeRef = database.child("challenges")
+    internal val dbUserRef = database.child("users")
+    internal val dbLikesRef = database.child("likes")
```

Comment:
` internal val dbLikesRef = database.child("likes")` seems like you create a `likes` table. I don't think we need that as it's not very helpful. The info about likes should be in `Challenge` and `User` as that would help you to retrieve the list of users that like a given challenge and a list of challenges that a given user likes. So more specifically what I mean is, there should be an attribute `likes` in `Challenge` just like `authorId`, `expirationDate`, etc. This `likes` attributes should be a list of users that like this given challenge. Similarly, in `User` there should be an attribute `likes` which would be the list of challenges that the user likes.

Reply:
I have changed the implementation of the firebase so that now each like is attached to a table under a given challenge in the firebase. This way the information about a like is not attached to a user but to a challenge, and so I can access the information about the user likes rapidly.

Reply:
Thank you for this comment. I agree very much with you. I have adapted the Firebase in such a way that the information about a given like is attached to a given user (so that a user has a list of challenges he or she likes) and also to a challenge (so that a given challenge has a list of references to the users that like this challenge). I think that this implementation will indeed work better in practice!